src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -326,6 +338,181 @@ bool check_decoder_model_compatibility(const std::shared_ptr<ov::Model>& decoder
+    class AttentionMaskInput : public ov::pass::MatcherPass {
+    public:
+        OPENVINO_RTTI("AttentionMaskInput");
+
+        AttentionMaskInput(std::shared_ptr<ov::Model> model) {
+            auto range = wrap_type<v4::Range>();
+            auto convert = wrap_type<v0::Convert>({range});
+            auto convert1 = wrap_type<v0::Convert>({convert});
+            auto greater = wrap_type<v1::Greater>({convert1, any_input()});
+            auto convert2 = wrap_type<v0::Convert>({greater});
+
+            register_matcher(std::make_shared<Matcher>(convert2, this->get_type_info().name), [model](Matcher& m) {
+                auto node = m.get_match_root();
+                auto attention_mask = std::make_shared<v0::Parameter>(ov::element::f32, ov::PartialShape{-1, -1});
+                attention_mask->get_output_tensor(0).set_names({"attention_mask"});
+                model->add_parameters({attention_mask});
+                ov::replace_node(node, attention_mask);
+                return false;
+            });
+        }
+    };
+
+    ov::pass::Manager pm;
+    pm.register_pass<AttentionMaskInput>(model);
+    pm.run_passes(model);
+}
+
+void add_attention_mask_input(std::shared_ptr<ov::Model> model) {
+    using namespace ov::pass::pattern;
+    using namespace ov::op;
+    class AttentionMaskInput : public ov::pass::MatcherPass {
+    public:
+        OPENVINO_RTTI("AttentionMaskInput");
+
+        AttentionMaskInput(std::shared_ptr<ov::Model> model) {
+            auto range = wrap_type<v4::Range>();
+            auto convert1 = wrap_type<v0::Convert>({range});
+            auto greater = wrap_type<v1::Greater>({convert1, any_input()});
+            auto convert2 = wrap_type<v0::Convert>({greater});
+
+            register_matcher(std::make_shared<Matcher>(convert2, this->get_type_info().name), [model](Matcher& m) {
+                auto node = m.get_match_root();
+                auto attention_mask = std::make_shared<v0::Parameter>(ov::element::f32, ov::PartialShape{-1, -1});
+                attention_mask->get_output_tensor(0).set_names({"attention_mask"});
+                model->add_parameters({attention_mask});
+                ov::replace_node(node, attention_mask);
+                return false;
+            });
+        }
+    };
+
+    ov::pass::Manager pm;
+    pm.register_pass<AttentionMaskInput>(model);
+    pm.run_passes(model);
+}
+
+void reshape_to_static(std::shared_ptr<ov::Model> model, const uint32_t input_size, const uint32_t kvcache_size) {
+    std::map<std::string, ov::PartialShape> new_shapes;
+    for (auto input : model->inputs()) {
+        const auto& input_name = input.get_any_name();
+        ov::PartialShape new_shape;
+        if (input_name.find("input_ids") != std::string::npos) {
+            new_shape = ov::PartialShape({1, input_size});
+        } else if (input_name.find("attention_mask") != std::string::npos) {
+            new_shape = ov::PartialShape({1, kvcache_size + 1});
```

Comment:
Since 448 is passed as kvcache_size, using kvcache_size + 1 creates attention_mask of size 1, 449 which is wrong as 448 is max supported by model. 

Use 1, kvcache_size for mask, as the past_key_values is later reshaped to kvcache_size - 1.

Reply:
I believe it's the matter of naming. The real kv cache size is 449 in this case though